python/ray/train/lightning/_lightning_utils.py

```diff
@@ -172,10 +188,25 @@ class RayModelCheckpoint(ModelCheckpoint):
     creates an AIR checkpoint whenever a lightning checkpoint is saved.
     """
 
-    def setup(self, *args, **kwargs) -> None:
-        super().setup(*args, **kwargs)
+    def setup(
+        self,
+        trainer: "pl.Trainer",
+        pl_module: "pl.LightningModule",
+        stage: Optional[str] = None,
+    ) -> None:
+        super().setup(trainer, pl_module, stage)
         self.is_checkpoint_step = False
 
+        if isinstance(trainer.strategy, DeepSpeedStrategy):
+            # For DeepSpeed, each node has a unique set of param and optimizer states,
+            # so the local rank 0 workers report the checkpoint shards for all workers
+            # on their node.
+            self.is_report_rank = session.get_local_rank() == 0
+        else:
+            # For DDP and FSDP, only the global rank 0 worker saves the full model.
```

Comment:
is that true for FSDP in all cases? I thought it depends on a setting

Reply:
Yes, one can switch to save one shard per workerfor FSDP. However, currently Lightning by default gathers all shards to Rank 0. We can update this logic after they support it.